### PR TITLE
allow globs for `denyImports` and `denyFiles`

### PR DESCRIPTION
**Pull request by pcattori**

### CODE REVIEW

File: .changeset/dirty-taxis-learn.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+"vite-env-only": minor
+---
+
+Allow globs for `denyImports` and `denyFiles`
+
+Using [micromatch][https://github.com/micromatch/micromatch] for pattern matching
```
Comment (markdalgleish) [inline .changeset/dirty-taxis-learn.md:7]: Out of curiosity, why'd you go with this library?

File: src/validate-id.ts [hunk 2]
```diff
@@ -25,7 +27,7 @@ export function validateId({
 
   for (const pattern of patterns) {
     if (
-      (typeof pattern === "string" && id === pattern) ||
+      (typeof pattern === "string" && isMatch(id, pattern)) ||
```
Comment (markdalgleish) [inline src/validate-id.ts:30]: Love how simple this change is!

File: .changeset/dirty-taxis-learn.md (hunk 1 above)
Comment (pcattori) [reply .changeset/dirty-taxis-learn.md:7]: Vite uses `fast-glob`, and `fast-glob` uses `minimatch`

Comment (markdalgleish) [reply .changeset/dirty-taxis-learn.md:7]: > `fast-glob` uses `micromatch`

Ah nice, I was missing this context 👍

**Review by markdalgleish — APPROVED** [review]

**Pull request merged**